vignettes/performance_analysis.Rmd

```diff
@@ -0,0 +1,323 @@
+  coords = c("lon", "lat"),
+  voter_id = "voter_id"
+)
+print(as.data.frame(voter_file_w_ward)[, c("surname", "WARD")])
+```
+
+We can double check that the correct wards were identified by plotting the voters on the ward map:
+
+```{r plot_voter_w_ward, fig.height=6, fig.width=7.2, fig.align = "center", message=FALSE, warning=FALSE}
+# Plot the map with no fill and voters as points
+options(repr.plot.width = 7.2, repr.plot.height = 6)
+map <- ggplot() +
+  geom_sf(data = ersd_maps, fill = "white") +
+  geom_sf_label(data = ersd_maps, aes(label = WARD), size = 3) +
+  geom_sf(data = voter_file_w_ward, size = 4, color = "black") +
+  xlab("Latitude") +
+  ylab("Longitude") +
+  theme_bw(base_size = 10) +
+  theme(
+    axis.title.x = element_text(size = 15, face = "bold", margin = margin(t = 5)),
+    axis.title.y = element_text(size = 15, face = "bold", margin = margin(r = 5))
+  )
+show(map)
+```
+This function can also be used to join the voter file to a shapefile of Census blocks, to facilitate predicting race with BISG. Since the Census shape file is too large to include in the package, we'll simply add the Census information by hand. However, the function would be used in exactly the same way, replacing `ersd_maps` with the name of the variable containing the Census shape.
+
+```{r merge_to_census}
+voter_file_w_ward$state <- rep("36", 5)
+voter_file_w_ward$county <- rep("087", 5)
+voter_file_w_ward$tract <- c("010801", "012202", "012501", "011502", "012202")
+voter_file_w_ward$block <- c("1016", "3002", "1016", "4001", "2004")
+```
+
+## Predicting race using BISG
+
+Since New York state does not report race on the voter file, we need to estimate it using BISG (see the BISG vignette for more details on this approach). Briefly, BISG provides a probabilistic estimate of race by combining knowledge of a voter's location and surname, both of which are informative of their race. `eiCompare` has a wrapper function for passing a voter file into the BISG function provided by the WRU package. To use this function, we'll load some Census data that was extracted using WRU containing information about the racial demographics of Rockland County:
+
+```{r apply_bisg}
+# Load Rockland County Census information
+data(rockland_census)
+
+# Apply BISG to the voter file to get race predictions
+bisg <- eiCompare::wru_predict_race_wrapper(
+  voter_file = as.data.frame(voter_file_w_ward),
+  census_data = rockland_census,
+  voter_id = "voter_id",
+  surname = "surname",
+  state = "NY",
+  county = "county",
+  tract = "tract",
+  block = "block",
+  census_geo = "block",
+  use_surname = TRUE,
+  surname_only = FALSE,
+  surname_year = 2010,
+  use_age = FALSE,
+  use_sex = FALSE,
+  return_surname_flag = TRUE,
+  return_geocode_flag = TRUE,
+  verbose = FALSE
+)
+```
+
+Let's take a look at the race probabilities:
+
+```{r race_probs}
+print(bisg$bisg[, c(
+  "voterid",
+  "surname",
+  "pred.whi",
+  "pred.bla",
+  "pred.his",
+  "pred.asi"
+)])
+```
```

Comment:
Maybe worth really slowly walking through how to interpret these probabilities, eg. "0.71 means the person has a 71% probability of being race X" but with better wording. Just so you have some description of this output before moving on to the next thing.